aws_batch/pipeline.py

```diff
@@ -798,23 +815,25 @@ def run_star(sample_name, fastq_file_1, fastq_file_2, star_genome_s3_path,
         execute_command("aws s3 cp %s %s/" % (star_genome_s3_path, REF_DIR))
         execute_command("cd %s; tar xvfz %s" % (REF_DIR, genome_file))
         logging.getLogger().info("downloaded index")
-    star_command_params = ['cd', scratch_dir, ';', STAR,
-                           '--outFilterMultimapNmax', '99999',
-                           '--outFilterScoreMinOverLread', '0.5',
-                           '--outFilterMatchNminOverLread', '0.5',
-                           '--outReadsUnmapped', 'Fastx',
-                           '--outFilterMismatchNmax', '999',
-                           '--outSAMmode', 'None',
-                           '--clip3pNbases', '0',
-                           '--readFilesCommand', 'zcat',
-                           '--runThreadN', str(multiprocessing.cpu_count()),
-                           '--genomeDir', REF_DIR + '/STAR_genome',
-                           '--readFilesIn', fastq_file_1, fastq_file_2]
-    execute_command(" ".join(star_command_params))
-    logging.getLogger().info("finished job")
-    # extract out unmapped files
-    execute_command("cp %s/%s %s/%s;" % (scratch_dir, 'Unmapped.out.mate1', result_dir, STAR_OUT1))
-    execute_command("cp %s/%s %s/%s;" % (scratch_dir, 'Unmapped.out.mate2', result_dir, STAR_OUT2))
+    # Check if parts.txt file exists, if so use the new version of (partitioned indices). Otherwise, stay put
+    if os.path.isfile("%s/STAR_genome/parts.txt" % REF_DIR):
+        num_parts = int(open("%s/STAR_genome/parts.txt" % REF_DIR, 'rb').read())
```

Comment:
are we worried about leaking file descriptors?   i guess this is not a long running process?

Reply:
sure. i can close it.